tests/test_calc_ab_diffs.py

```diff
@@ -17,22 +17,66 @@
 def test_calc_record_diff_has_diff():
     a = {"color": "green"}
     b = {"color": "red"}
-    assert calc_record_diff(json.dumps(a).encode(), json.dumps(b).encode()) == json.dumps(
-        {"color": ["green", "red"]}
+    ab_diff, modified_timdex_fields, has_diff = calc_record_diff(a, b)
+    assert ab_diff == json.dumps(
+        {"values_changed": {"root['color']": {"new_value": "red", "old_value": "green"}}}
     )
+    assert modified_timdex_fields == {"color"}
+    assert has_diff
 
 
 def test_calc_record_diff_no_diff():
     a = {"color": "green"}
     b = a
-    assert calc_record_diff(json.dumps(a).encode(), json.dumps(b).encode()) == json.dumps(
-        {}  # no diff
-    )
+    ab_diff, modified_timdex_fields, has_diff = calc_record_diff(a, b)
+    assert ab_diff == json.dumps({})  # no diff
+    assert not modified_timdex_fields
+    assert not has_diff
 
 
 def test_calc_record_diff_one_input_is_none():
     a = {"color": "green"}
-    assert calc_record_diff(json.dumps(a).encode(), None) is None
+    assert calc_record_diff(a, None) == (None, None, False)
+
+
+def test_calc_record_diff_array_by_default_order_not_a_diff():
+    """Arrays with the same values, but differently ordered, not considered a diff."""
+    a = {"colors": ["green", "red"]}
+    b = {"colors": ["red", "green"]}
+    ab_diff, modified_timdex_fields, has_diff = calc_record_diff(a, b)
+    assert ab_diff == json.dumps({})  # no diff
+    assert not modified_timdex_fields
+    assert not has_diff
```

Comment:
Loving that we can have this test already, I thought it was going to be a far more involved fix